eldritch/gate_encounters.py

```diff
@@ -20,64 +23,393 @@ def encounter_event(self, character, world_name):
     return self.encounters[world_name](character)
 
 
-def Dreamlands10(char):
+def Abyss1(char) -> events.Event:
+  check = events.Check(char, "luck", -1)
+  succeed = events.Return(char, char.place.info.name)
+  monster = events.MonsterAppears(char)
+  return events.PassFail(char, check, succeed, monster)
+
+
+def GreatHall1(char) -> events.Event:
+  check = events.Check(char, "sneak", -2)
+  spell = events.Draw(char, "spells", float("inf"))
+  stamina = events.Loss(char, {"stamina": 2})
+  return events.PassFail(char, check, spell, stamina)
+
+
+def Other1(char) -> events.Event:
+  check = events.Check(char, "fight", -1)
+  spell = events.Draw(char, "spells", 1)
+  return events.PassFail(char, check, spell, events.Nothing())
+
+
+def Abyss2(char) -> events.Event:
+  return events.Gain(char, {"stamina": 1})  # TODO: Should we attach a source somehow?
+
+
+def Pluto2(char) -> events.Event:
+  check = events.Check(char, "fight", -2)
+  return events.PassFail(char, check, events.Nothing(), events.LostInTimeAndSpace(char))
+
+
+def Other2(char) -> events.Event:
+  # check = events.Check(char, "speed", -2)
+  # return events.Sequence(
+  #     [
+  #       events.PassFail(
+  #           char,
+  #           check,
+  #           events.Return(char, char.place.info.name),
+  #           events.LostInTimeAndSpace(char),
+  #       ),
+  #       events.CloseGate(char, "the one you entered"),
```

Comment:
I think this is the real showstopper for this event... Or does it really just mean close the gate you returned through? Because it's possible that the gate you entered the other world through is already closed, right?